Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -89,24 +97,32 @@ extension HTTPClient {
                 throw HTTPClientError.emptyHost
             }
 
-            self.version = version
-            self.method = method
-            self.url = url
-            self.scheme = scheme
-            self.host = host
-            self.headers = headers
-            self.body = body
+            self._url = url
+            self._schema = scheme
+            self._host = host
```

Comment:
Do we need to recalculate these? Unless we have a strong performance argument for doing so, it seems like it'd be better for `scheme` and `host` to be computed properties.

Reply:
two reasons:

=> if we use computed properties they will need to force unwrap since `host` and `scheme` are optional on `URL`. this means we rely on the fact we validated they are not empty in the url setter, which we do now. so could work. having said that, in my experience this would be a fragile design as someone in the future may decide to change the validation and forget there are computed properties that make this assumption leading to runtime errors. the current implementation makes this very explicit and hard to get wrong in the future 

=> we are lowercasing the `scheme` which is used for business logic and validation downstream code. using the url's schema will not include this manipulation and can cause business logic issues. we could instead create a new url object with the lowercased schema but the current implementation seems less intrusive and expensive

Reply:
i could go either way on this one,  no strong opinion. @artemredkin @ianpartridge @tanner0101 ?

Reply:
Tbh I don't think it's worth spending the cycles discussing it. Let's leave it as is, fixing the typo of `schema` to `scheme`.